### ISSUE TITLE
Should we have import of Engine in ignite/__init__.py?

### ISSUE THREAD

**Issue by alykhantejani** [pytorch/ignite#179]
@elanmart thanks for the DCGAN PR -  looks great! 

Just noticed in the PR that 
```
from ignite.engine.engine import Engine, Events, State
``` 
was added to `ignite/__init__.py`?

I understand that it does add some convenience so users can just use the Engine, but this also creates inconsistent usage patterns i.e. 

```
import ignite
engine = ignite.Engine(...)
```
and 
```
from ignite.engine import Engine, State, Events
engine = Engine()
```

perhaps we could just add 
```
import ignite.engine
import ignite.handlers
```
etc. in the `__init__.py` file in `ignite/`, similar to what's done in pytorch (i.e. [here](https://github.com/pytorch/pytorch/blob/master/torch/__init__.py#L276))

cc @vfdev-5

**Comment by alykhantejani** [comment]
Copying some bits over from the slack discussion:

PEP8 says:
```
Imported names should always be considered an implementation detail. Other modules must not rely on indirect access to such imported names unless they are an explicitly documented part of the containing module's API, such as os.path or a package's __init__ module that exposes functionality from submodules.
```
So going by this we probably shouldn’t allow users to rely on the indirect import from __init__

It’s not a huge deal to keep it in, but I think we should only keep it in if it adds some benefit, otherwise the inconsistent patterns are not great stylistically

cc @jasonkriss in case you have an opinion on this

**Comment by jasonkriss** [comment]
@alykhantejani completely agreed.

**Comment by vfdev-5** [comment]
Right, so we can change the imports of `ignite/__init__.py` to 
```python
import ignite.engine
import ignite.handlers
```

**Comment by alykhantejani** [comment]
Merged in #180
